### PR TITLE
Add docker stuff, and use it in Travis CI. Update scripts as needed

### LINKED ISSUE #281: Add CentOS packaging support
At least for CentOS 8, and possibly CentOS 7 as well.

### LINKED ISSUE #286: Create Dockerfile(s) for the project
This would be so that we, and others, could use Docker images to build and package Vega Strike. We would probably end up doing this in our CI/CD as well.

### LINKED ISSUE #292: Add Debian 9 stretch packaging support

### LINKED ISSUE #294: Add Fedora 30 to Travis CI

### PR DESCRIPTION
**Pull request by stephengtuggy**
Thank you for submitting a pull request and becoming a contributor to the Vega Strike Core Engine.

Please answer the following:

Code Changes:
- [x] Have the PR Validation Tests been run? See https://github.com/vegastrike/Vega-Strike-Engine-Source/wiki/Pull-Request-Validation
- [ ] This is a documentation change only

Issues:
- Please list any related issues
**Closes #286**
**Closes #281**
**Closes #294**
**Closes #292**

Purpose:
- What is this pull request trying to do? **Dockerize the project. Use Docker to build for more distros on Travis CI. Etc.**
- What release is this for? **0.7.x, probably**
- Is there a project or milestone we should apply this to? **0.7.x, probably**

### CODE REVIEW

File: engine/src/game_config.h [hunk 1]
```diff
@@ -19,28 +19,48 @@ using pt::ptree;
...
+        string const key = section + "." + name;
+        if (variables.count(key))
+            return variables[key];
+        return DEFAULT_ERROR_VALUE;
+    }
 public:
     static void LoadGameConfig(const string &filename);
     template <class T>
-    static T GetVariable(string const &section, string const &name, T default_value) = delete;
+    static inline T GetVariable(string const &section, string const &name, T default_value) = delete;
```
Comment (stephengtuggy) [inline engine/src/game_config.h:36]: Xenial from Docker has an older version of clang. 3.8, I believe. clang 3.8 on Xenial apparently doesn't handle well `= delete` template methods with specializations. See https://stackoverflow.com/a/33258249/5067822 . For this reason, I have moved `Xenial clang` to Allowed Failures on Travis.

Comment (BenjamenMeyer) [reply engine/src/game_config.h:36]: @stephengtuggy does it help to do a full upgrade on the docker image:
```
# apt-get update
# apt-get -y upgrade
# apt-get -y dist-upgrade
```

FYI - created https://hub.docker.com/orgs/vegastrike for you

**Comment by stephengtuggy** [comment]
I did make a few changes to the actual code, so this should be play-tested. Haven't had a chance to do that yet myself. If someone else wants to, that would be great.

**Comment by stephengtuggy** [comment]
PR Validation Tests now complete.

**Pull request merged**